endpoints/main.go

```diff
@@ -28,17 +33,28 @@ var totalFailedUptimeChecksCounterVec *prometheus.CounterVec
 // Record the metrics into the appropriate prometheus register under the label specified by organizationName
 // recordLongRunningMetrics specifies wether or not to record information contained in the capability statment
 func getHTTPRequestTiming(urlString string, organizationName string, recordLongRunningMetrics bool) {
-	var resp, responeTime, err = querier.GetResponseAndTiming(urlString)
+	ctx := context.Background()
+	// Drop connection if no reply within 30 seconds
+	ctx, cancelFunc := context.WithDeadline(ctx, time.Now().Add(30*time.Second))
+	var resp, responseTime, err = querier.GetResponseAndTiming(ctx, urlString)
+
+	if cancelFunc != nil {
```

Comment:
remove this chunk - this made sense when the `cancelFunc` was a return of `GetResponseAndTiming`, but now that it's a part of `WithDeadline`, I don't believe you need to check the `nil` value and that best practice is to do the `defer` immediately.